src/client/common/installer/productInstaller.ts

```
@@ -346,6 +352,46 @@ export class RefactoringLibraryInstaller extends BaseInstaller {
 }
 
 export class DataScienceInstaller extends BaseInstaller {
+    // Override base installer to support a more DS-friendly streamlined installation.
+    public async install(
+        product: Product,
+        interpreterUri?: InterpreterUri,
+        cancel?: CancellationToken
+    ): Promise<InstallerResponse> {
+        // Get a list of known installation channels, pip, conda, etc.
+        const channels: IModuleInstaller[] = await this.serviceContainer
+            .get<IInstallationChannelManager>(IInstallationChannelManager)
+            .getInstallationChannels();
+
+        // Assume the Uri is actually a full PythonInterpreter...
+        const interpreter = interpreterUri as PythonInterpreter;
```

Honestly I sometimes find the usage of InterpreterUri a bit confusing as seems to represent two different concepts to me. But currently the interface can be passed an Interpreter or a URI here. Does it make sense if passed a URI to just get the interpreter for that URI then run the rest of this code as normal with that interpreter? That way we don't just return an InstallerError.Ignore when what looks like a valid usage of the interpreter is called with a URI not an interpreter.

---

@DonJayamanne I feel like you might know that better. Would that make sense?

---

Good point, 

>  Does it make sense if passed a URI to just get the interpreter for that URI then run the rest of this code as normal with that interpreter?

I think if the requirement is for an interpreter to be passed in when using DS functionality, then we can throw an error (I've added comments), ie. we shouldn't be returning `Ignore` to silently ignore such issues.